File: src/main/java/com/example/concetto/models/Concept.java
package com.example.concetto.models;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.Date;

@Data
@Entity
public class Concept {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne()
    @JoinColumn(name="user_id", nullable = false)
    @NotNull
    private User user;

    @ManyToOne()
    @JoinColumn(name="subject_id", nullable = false)
    @NotNull
    private Subject subject;

    @OneToOne()
    @JoinColumn(name="inter_interval_id", nullable = false)
    @NotNull
    private InterInterval interInterval;

    @NotBlank
    @Size(max = 100)
    private String name;

    @Size(max = 255)
    private String explanation;

    private boolean reviewed;
    private boolean simplified;
    private boolean done;
    private Date dateCreated;
    private Date nextReviewDate;

    public Concept() {
        this.dateCreated = new Date();
        this.nextReviewDate = new Date();
    }
}

File: client/src/app/models/concept.ts
import {Subject} from './subject';
import { InterInterval } from './interInteval';
import { InterIntervalService } from '../services/inter-interval.service';

export class Concept {
    id : number;
    subject : Subject;
    name : string;
    explanation : string;
    reviewed: boolean;
    simplified: boolean;
    done : boolean;
    interInterval : InterInterval;

    constructor(){
        this.name = "";
        this.explanation = "";
        this.reviewed = false;
        this.simplified = false;
    }

    get Id() : number {
        return this.id;
    }

    set Id(id : number) {
        this.id = id;
    }

    get Subject() : Subject {
        return this.subject;
    }

    set Subject(subject : Subject) {
        this.subject = subject;
    }

    get InterInterval() : InterInterval {
        return this.interInterval;
    }

    set InterInterval(interInteval : InterInterval) {
        this.interInterval = interInteval;
    }



    get Name() : string {
        return this.name;
    }

    set Name(name : string){
        this.name = name;
    }

    get Explanation() : string {
        return this.explanation;
    }

    set Explanation(explanation : string){
        this.explanation = explanation;
    }

    get Reviewed() : boolean {
        return this.reviewed;
    }

    set Reviewed(reviewed : boolean) {
        this.reviewed = reviewed;
    }

    get Simplified() : boolean {
        return this.simplified;
    }

    set Simplified(simplified : boolean) {
        this.simplified = simplified;
    }
}